### ISSUE TITLE
Game loading page freezes in Chrome and Chromium

### ISSUE THREAD

**Issue by montuos** [ChildrenOfUr/cou-issues#2459]
https://play.childrenofur.com/ doesn't load; the page freezes before you can even type in your name, let alone your password.  It stops loading just after the world starts spinning.  This has been happening since sometime yesterday or the day before.

It fails this way in both Chromium [Version 60.0.3112.113 (Developer Build) Built on Ubuntu , running on Ubuntu 16.04 (64-bit)] and Chrome [Version 60.0.3112.113 (Official Build) (64-bit)].  It doesn't even load that far in Vivaldi Snapshot [1.5.633.16 (Developer Build) dev (64-bit)]; the screen stays completely blank in that browser.
However, the game loads just fine in both Firefox [55.0.2 (64-bit)] and Vivaldi stable [1.11.917.43 (Stable channel) (64-bit)].
I'm running Ubuntu MATE if that makes any difference.

![coufreeze20170831](https://user-images.githubusercontent.com/7076290/29942810-241cf93e-8e65-11e7-95f8-c3d3149c2f0e.png)

**Comment by Klikini** [comment]
Is this still an issue?

**Comment by montuos** [comment]
Yes, it is.  The world starts spinning, the world stops spinning, the music keeps playing, and I can't even click into the email or password fields.  Vivaldi Snapshot still remains a completely blank page.

Current versions:
Bad
Chromium 62.0.3202.75 (Official Build)
Chrome 62.0.3202.89 (Official Build) (64-bit)
Vivaldi Snapshot 1.5.633.16 (Developer Build) dev (64-bit).

Good
Firefox 56.0 (64-bit)
Vivaldi Stable 1.12.955.48 (Stable channel) (64-bit)

OS version is still Ubuntu MATE 16.04 LTS.

**Comment by Klikini** [comment]
We do not support browser snapshots/betas/pre-releases/etc, so I'm not worried about Vivaldi until that code makes it to their stable version.

We definitely support Linux (our server runs on Ubuntu, and most of the devs use Linux at least part of the time, making it probably the most extensively tested OS), so you don't have to worry about that part.

As for Chromium and Chrome not working: that's really weird, because we test in them most often. Have you tried using incognito mode?

**Comment by montuos** [comment]
Yes, it's exactly the same in incognito windows, and yes, I did try with every single one of my extensions disabled.

So now I'm wondering whether there is some setting I have turned off that is necessary for it to work (or on that needs to be off).

**Comment by Klikini** [comment]
I suggested incognito mode because that usually works for users who can't log in because of cookies/local storage/etc, I didn't even think about extensions...

Can you see if there are any errors in the Chrome and/or Chromium dev console?

**Comment by montuos** [comment]
Ok, I figured out how to open the console and finally got logs, but I have no idea how to recognize whether there are any actual errors here.

When I had the console open with CoU, both Chrome and Chromium crashed so hard I couldn't either close the tab or quit normally like before; I actually had to open the terminal to kill all the processes.

Here's the contents of the log I saved from Chrome (Chromium's follows):

17:33:41.814 newtab:8 SW registered
17:34:05.281 Navigated to https://play.childrenofur.com/
17:34:05.503 [Deprecation] /deep/ combinator in CSS is deprecated and will be removed in M63, around December 2017. See https://www.chromestatus.com/features/4964279606312960 for more details.
17:34:05.540 (index):6065 Respecting Do-Not-Track request, disabling Google Analytics.
17:34:07.354 polymer.min.js:14 [Deprecation] Calling Element.createShadowRoot() for an element which already hosts a shadow root is deprecated and will be removed in M63, around December 2017. See https://www.chromestatus.com/features/4668884095336448 for more details.
shadowFromTemplate @ polymer.min.js:14
parseDeclaration @ polymer.min.js:14
parseDeclarations @ polymer.min.js:14
makeElementReady @ polymer.min.js:14
createdCallback @ polymer.min.js:14
newCallback @ interop_support.js:38
newCallback @ interop_support.js:38
registerElement @ interop_support.js:64
registerPrototype @ polymer.min.js:14
register @ polymer.min.js:14
(anonymous) @ index.html.polymer.bootstrap.dart.js:41187
dart.bcR.$3 @ index.html.polymer.bootstrap.dart.js:51381
kf @ index.html.polymer.bootstrap.dart.js:1916
dart.Pp @ index.html.polymer.bootstrap.dart.js:41069
(anonymous) @ index.html.polymer.bootstrap.dart.js:41189
_register @ polymer.min.js:14
flush @ polymer.min.js:14
ready @ polymer.min.js:14
check @ polymer.min.js:14
(anonymous) @ polymer.min.js:14
n @ webcomponents.min.js:13
i @ webcomponents.min.js:13
(anonymous) @ webcomponents.min.js:13
r @ webcomponents.min.js:13
t @ webcomponents.min.js:13
(anonymous) @ polymer.min.js:14
c @ polymer.min.js:13
characterData (async)
b @ polymer.min.js:13
d @ polymer.min.js:14
(anonymous) @ index.html.polymer.bootstrap.dart.js:41148
dart.NI.i.b0E @ index.html.polymer.bootstrap.dart.js:53054
dart.bcP @ index.html.polymer.bootstrap.dart.js:50372
dart.bkt.$1 @ index.html.polymer.bootstrap.dart.js:51360
dart.aFY.i2 @ index.html.polymer.bootstrap.dart.js:21589
dart.eS.aT2.H.o.b.func @ index.html.polymer.bootstrap.dart.js:19081
dart.b9L.$0 @ index.html.polymer.bootstrap.dart.js:19317
dart.Z.i.un @ index.html.polymer.bootstrap.dart.js:19246
dart.Z.Bv.H.o.a.func @ index.html.polymer.bootstrap.dart.js:19178
dart.b9F.$0 @ index.html.polymer.bootstrap.dart.js:19286
dart.bd8 @ index.html.polymer.bootstrap.dart.js:18408
dart.c1D @ index.html.polymer.bootstrap.dart.js:18410
dart.b8A.$1 @ index.html.polymer.bootstrap.dart.js:18560
dart.bh9.$0 @ index.html.polymer.bootstrap.dart.js:2693
xX @ index.html.polymer.bootstrap.dart.js:1310
Bc @ index.html.polymer.bootstrap.dart.js:1058
dart.bh7 @ index.html.polymer.bootstrap.dart.js:2034
(anonymous) @ index.html.polymer.bootstrap.dart.js:2042
childList (async)
dart.b8z.$1 @ index.html.polymer.bootstrap.dart.js:18567
dart.aHO @ index.html.polymer.bootstrap.dart.js:18416
dart.PJ @ index.html.polymer.bootstrap.dart.js:18510
dart.aFY.j8 @ index.html.polymer.bootstrap.dart.js:21597
dart.Z.cA @ index.html.polymer.bootstrap.dart.js:19188
dart.uG @ index.html.polymer.bootstrap.dart.js:45128
(anonymous) @ index.html.polymer.bootstrap.dart.js:55249
dart.aJ8 @ index.html.polymer.bootstrap.dart.js:50367
dart.bh0 @ index.html.polymer.bootstrap.dart.js:50365
dart.Qb @ index.html.polymer.bootstrap.dart.js:51559
dart.cfT @ index.html.polymer.bootstrap.dart.js:3348
xX @ index.html.polymer.bootstrap.dart.js:1310
aJ9 @ index.html.polymer.bootstrap.dart.js:1102
(anonymous) @ index.html.polymer.bootstrap.dart.js:67406
(anonymous) @ index.html.polymer.bootstrap.dart.js:67406
init.currentScript @ index.html.polymer.bootstrap.dart.js:67401
(anonymous) @ index.html.polymer.bootstrap.dart.js:67404
(anonymous) @ index.html.polymer.bootstrap.dart.js:67406
17:34:08.267 index.html.polymer.bootstrap.dart.js:42837 [SoundManager] Starting up... 0:00:00
17:34:09.439 index.html.polymer.bootstrap.dart.js:42837 [Server Communication] Map data loaded. 0:00:01

And here's the log from Chromium:

[Deprecation] /deep/ combinator in CSS is deprecated and will be removed in M63, around December 2017. See https://www.chromestatus.com/features/4964279606312960 for more details.
polymer.min.js:14 [Deprecation] Calling Element.createShadowRoot() for an element which already hosts a shadow root is deprecated and will be removed in M63, around December 2017. See https://www.chromestatus.com/features/4668884095336448 for more details.
shadowFromTemplate @ polymer.min.js:14
parseDeclaration @ polymer.min.js:14
parseDeclarations @ polymer.min.js:14
makeElementReady @ polymer.min.js:14
createdCallback @ polymer.min.js:14
newCallback @ interop_support.js:38
newCallback @ interop_support.js:38
registerElement @ interop_support.js:64
registerPrototype @ polymer.min.js:14
register @ polymer.min.js:14
(anonymous) @ index.html.polymer.bootstrap.dart.js:41187
dart.bcR.$3 @ index.html.polymer.bootstrap.dart.js:51381
kf @ index.html.polymer.bootstrap.dart.js:1916
dart.Pp @ index.html.polymer.bootstrap.dart.js:41069
(anonymous) @ index.html.polymer.bootstrap.dart.js:41189
_register @ polymer.min.js:14
flush @ polymer.min.js:14
ready @ polymer.min.js:14
check @ polymer.min.js:14
(anonymous) @ polymer.min.js:14
n @ webcomponents.min.js:13
i @ webcomponents.min.js:13
(anonymous) @ webcomponents.min.js:13
r @ webcomponents.min.js:13
t @ webcomponents.min.js:13
(anonymous) @ polymer.min.js:14
c @ polymer.min.js:13
characterData (async)
b @ polymer.min.js:13
d @ polymer.min.js:14
(anonymous) @ index.html.polymer.bootstrap.dart.js:41148
dart.NI.i.b0E @ index.html.polymer.bootstrap.dart.js:53054
dart.bcP @ index.html.polymer.bootstrap.dart.js:50372
dart.bkt.$1 @ index.html.polymer.bootstrap.dart.js:51360
dart.aFY.i2 @ index.html.polymer.bootstrap.dart.js:21589
dart.eS.aT2.H.o.b.func @ index.html.polymer.bootstrap.dart.js:19081
dart.b9L.$0 @ index.html.polymer.bootstrap.dart.js:19317
dart.Z.i.un @ index.html.polymer.bootstrap.dart.js:19246
dart.Z.Bv.H.o.a.func @ index.html.polymer.bootstrap.dart.js:19178
dart.b9F.$0 @ index.html.polymer.bootstrap.dart.js:19286
dart.bd8 @ index.html.polymer.bootstrap.dart.js:18408
dart.c1D @ index.html.polymer.bootstrap.dart.js:18410
dart.b8A.$1 @ index.html.polymer.bootstrap.dart.js:18560
dart.bh9.$0 @ index.html.polymer.bootstrap.dart.js:2693
xX @ index.html.polymer.bootstrap.dart.js:1310
Bc @ index.html.polymer.bootstrap.dart.js:1058
dart.bh7 @ index.html.polymer.bootstrap.dart.js:2034
(anonymous) @ index.html.polymer.bootstrap.dart.js:2042
childList (async)
dart.b8z.$1 @ index.html.polymer.bootstrap.dart.js:18567
dart.aHO @ index.html.polymer.bootstrap.dart.js:18416
dart.PJ @ index.html.polymer.bootstrap.dart.js:18510
dart.aFY.j8 @ index.html.polymer.bootstrap.dart.js:21597
dart.Z.cA @ index.html.polymer.bootstrap.dart.js:19188
dart.uG @ index.html.polymer.bootstrap.dart.js:45128
(anonymous) @ index.html.polymer.bootstrap.dart.js:55249
dart.aJ8 @ index.html.polymer.bootstrap.dart.js:50367
dart.bh0 @ index.html.polymer.bootstrap.dart.js:50365
dart.Qb @ index.html.polymer.bootstrap.dart.js:51559
dart.cfT @ index.html.polymer.bootstrap.dart.js:3348
xX @ index.html.polymer.bootstrap.dart.js:1310
aJ9 @ index.html.polymer.bootstrap.dart.js:1102
(anonymous) @ index.html.polymer.bootstrap.dart.js:67406
(anonymous) @ index.html.polymer.bootstrap.dart.js:67406
init.currentScript @ index.html.polymer.bootstrap.dart.js:67401
(anonymous) @ index.html.polymer.bootstrap.dart.js:67404
(anonymous) @ index.html.polymer.bootstrap.dart.js:67406
index.html.polymer.bootstrap.dart.js:42837 [SoundManager] Starting up... 0:00:00
index.html.polymer.bootstrap.dart.js:42837 [Server Communication] Map data loaded. 0:00:05

**Comment by Klikini** [comment]
Both of those logs look normal. What are your system specs? If it runs out of memory before it can load all the assets, Chrome might consider it frozen and stop running code.

**Comment by montuos** [comment]
I've got 16 GB of RAM.  But why would Chrome and Chromium run out of memory when Firefox and Vivaldi don't?
